### ISSUE TITLE
Crashes when trying to repack a/0/9/4 GARC for Pokémon Moon (runs out of ram for 32 bit process)

### ISSUE THREAD

**Issue by ApacheThunder** [kwsch/pk3DS#42]
Tried to use the folder repacker part of the garc tool to repack the a/0/9/4_g folder.

however it crashes with this:

Problem Event Name:	CLR20r3
  Problem Signature 01:	pk3DS.exe
  Problem Signature 02:	1.0.0.0
  Problem Signature 03:	583df1f2
  Problem Signature 04:	mscorlib
  Problem Signature 05:	4.6.1055.0
  Problem Signature 06:	563c0eac
  Problem Signature 07:	157f
  Problem Signature 08:	1d7
  Problem Signature 09:	System.IO.IOException
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

EDIT: After further testing it appears pk3DS failed to exit properly a few times and caused the folder to be inaccessible to further instances of pk3DS which caused the crash. I had to manually kill the duplicate processs with task manager before it would pack.

EDIT2: It appears that the first couple times I attempted it, I didn't notice that it was actually packing the folder (as the progress bar appeared). But it doesn't disable any other controls while it's doing this hence why I encountered wierd bugs/crashes trying to pack it a second time. Controls should probably be locked out until it finishes packing a folder.

**Comment by ApacheThunder** [comment]
Ok it still crashes. However this time it will crash after reaching 99% (impacting the same GARC file I'm trying to repack)

Problem signature:
  Problem Event Name:	CLR20r3
  Problem Signature 01:	pk3DS.exe
  Problem Signature 02:	1.0.0.0
  Problem Signature 03:	583df1f2
  Problem Signature 04:	mscorlib
  Problem Signature 05:	4.6.1055.0
  Problem Signature 06:	563c0eac
  Problem Signature 07:	17d6
  Problem Signature 08:	57
  Problem Signature 09:	System.OutOfMemoryException
  OS Version:	6.1.7601.2.1.0.256.1
  Locale ID:	1033
  Additional Information 1:	0a9e
  Additional Information 2:	0a9e372d3b4ad19135b953a78882e789
  Additional Information 3:	0a9e
  Additional Information 4:	0a9e372d3b4ad19135b953a78882e789

**Comment by ApacheThunder** [comment]
Adding 64bit to the available platforms in configuration manager seems to resolve this issue as it is now able to build this large garc file. After further testing I suggest you make this change as well and perhaps provide a 64bit exe in the next release. I guess this wasn't a problem until Sun/Moon came out. The garc that contains the pokémon battle models/textures is rather large. :P

**Comment by sebastienrouget** [comment]
Do you have over 2GB to repack ? If you do, then the problem may be the use of a [MemoryStream](https://github.com/kwsch/pk3DS/blob/master/pk3DS/3DS/GARC.cs#L197-L198), which is limited to a 2GB buffer in 32bit.
I suggest @kwsch to use a FileStream instead, and write directly to a file, to keep releasing in 32bit. Additionally to @ApacheThunder solution.

**Comment by kwsch** [comment]
I don't think anything is using more than 1GB of data for a stream; the largest file in S/M is a094 @ 1.3GB which is not modified by pk3DS. Next largest is 340MB.

relevant line:
`                GARCdata.CopyTo(newGARC);       // Copy the data.`

if there's 2 streams with 1.3GB I can see it tripping there (>2GB total), however, this wasn't an issue for past games when repacking the models. Are you sure there aren't any further-unpacked files in the garc you are trying to repack?

**Comment by ApacheThunder** [comment]
The GARC it crashes on is the one for the Pokémin ingame models. (it's over 1.2GB). I found it repacks this garc much faster if I skip decompression. (I haven't checked it crashes in this instance since i have have moved to using the 64bit exe I compiled). But the bin files in this GARC only contain textures/pokémon models/skeletons/animations, so there's nothing else in this garc that it's attempting to repack....

I guess it's only an issue with the garc part of the unpacker/packer tool in the tools menu. The main functions for game modifying aren't effected by this right now. But perhaps fix this issue to future proof things? Thus far I have not noticed any issues using the 64bit compiled exe.
